src/main/src/test/java/org/geoserver/security/workspaceadmin/WorkspaceAdminAuthorizationManagerTest.java

```diff
@@ -0,0 +1,132 @@
+        authSupplier = () -> authentication;
+        authorizer = mock(WorkspaceAdminAuthorizer.class);
+
+        // Setup WorkspaceAdminAuthorizer.get() to return our mock
+        GeoServerExtensionsHelper.singleton("workspaceAdminAuthorization", authorizer, WorkspaceAdminAuthorizer.class);
+
+        authManager = new WorkspaceAdminAuthorizationManager(metadataSource);
+
+        request = mock(HttpServletRequest.class);
+        when(request.getMethod()).thenReturn("GET");
+        when(request.getContextPath()).thenReturn("/geoserver");
+    }
+
+    @After
+    public void tearDown() {
+        GeoServerExtensionsHelper.init(null);
+    }
+
+    @Test
+    public void testCheckWhenAuthorizerNotAvailable() {
+        // Test when WorkspaceAdminAuthorizer is not available
+        GeoServerExtensionsHelper.clear();
+
+        // Should abstain when authorizer not available
+        AuthorizationDecision decision = authManager.check(authSupplier, request);
+        assertEquals(ACCESS_ABSTAIN, decision);
+    }
+
+    @Test
+    public void testCheckWhenNotFullyAuthenticated() {
+        // Test when user is not fully authenticated
+        when(authorizer.isFullyAuthenticated(authentication)).thenReturn(false);
+
+        // Should deny access when not fully authenticated
+        AuthorizationDecision decision = authManager.check(authSupplier, request);
+        assertEquals(ACCESS_DENIED, decision);
+    }
+
+    @Test
+    public void testCheckWhenNoMatchingRules() {
+        when(request.getRequestURI()).thenReturn("/geoserver/rest/logging");
+
+        // Setup authentication and request
+        when(authorizer.isFullyAuthenticated(authentication)).thenReturn(true);
+
+        // No matching rules
+        when(metadataSource.getAttributes(request)).thenReturn(Collections.emptyList());
+
+        // Should abstain when no matching rules
+        AuthorizationDecision decision = authManager.check(authSupplier, request);
+        assertEquals(ACCESS_ABSTAIN, decision);
+    }
+
+    @Test
+    public void testCheckWhenMatchingRuleButNotWorkspaceAdmin() {
+        // Setup authentication and request
+        when(authorizer.isFullyAuthenticated(authentication)).thenReturn(true);
+
+        // Create a matching rule
+        WorkspaceAdminRestAccessRule rule = new WorkspaceAdminRestAccessRule(1, "/rest/workspaces/topp", Set.of(GET));
+        when(metadataSource.getAttributes(request)).thenReturn(List.of(rule));
+        when(request.getRequestURI()).thenReturn("/geoserver/rest/workspaces/topp");
+
+        // But user is not a workspace admin
+        when(authorizer.isWorkspaceAdmin(authentication)).thenReturn(false);
+
+        // Should abstain when matching rule but not workspace admin
+        AuthorizationDecision decision = authManager.check(authSupplier, request);
+        assertEquals(ACCESS_ABSTAIN, decision);
+    }
+
+    @Test
+    public void testCheckWhenMatchingRuleAndWorkspaceAdmin() {
```

Comment:
Would it make sense to also test the case of a workspace admin but there's no matching rule?